src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -925,20 +936,60 @@ export default class ResultsViewOncoprint extends React.Component<
         invoke: async () => getUnalteredUids(this.geneticTracks.result!),
     });
 
-    readonly treatmentSelectOptions = remoteData<ISelectOption[]>({
-        await: () => [this.props.store.treatmentsInStudies],
-        invoke: async () =>
-            treatmentsToSelectOptions(
-                this.props.store.treatmentsInStudies.result || []
-            ),
+    readonly genericAssayEntitiesSelectOptionsGroupByGenericAssayType = remoteData<{
+        [genericAssayType: string]: ISelectOption[];
+    }>({
+        await: () => [
+            this.props.store.genericAssayEntitiesGroupByGenericAssayType,
+        ],
+        invoke: async () => {
+            return Promise.resolve(
+                genericAssayEntitiesToSelectOptionsGroupByGenericAssayType(
+                    this.props.store.genericAssayEntitiesGroupByGenericAssayType
+                        .result || {}
+                )
+            );
+        },
+    });
+
+    readonly genericAssayEntitiesSelectOptionsGroupByMolecularProfileId = remoteData<{
+        [genericAssayType: string]: ISelectOption[];
+    }>({
+        await: () => [
+            this.props.store.genericAssayEntitiesGroupByMolecularProfileId,
+        ],
+        invoke: async () => {
+            return Promise.resolve(
+                genericAssayEntitiesToSelectOptionsGroupByGenericAssayType(
+                    this.props.store
+                        .genericAssayEntitiesGroupByMolecularProfileId.result ||
```

Comment:
again, i don't think that the result of this promise can return as something that would evaluate to false (false|udefined|null) as long as the await is complete.   please verify that.  this is mostly just code clarity issue.  i.e. i don't think you need do this check.   but i could be wrong.